solr/solrj/src/test/org/apache/solr/client/solrj/request/TestUpdateRequest.java

```diff
@@ -17,52 +17,164 @@
 package org.apache.solr.client.solrj.request;
 
 import java.util.Arrays;
+import java.util.List;
 
+import com.google.common.collect.Lists;
 import org.apache.solr.common.SolrInputDocument;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
 
+import static org.apache.solr.SolrTestCaseJ4.adoc;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
 public class TestUpdateRequest {
 
   @Rule
   public ExpectedException exception = ExpectedException.none();
 
   @Before
   public void expectException() {
-    exception.expect(NullPointerException.class);
-    exception.expectMessage("Cannot add a null SolrInputDocument");
   }
 
   @Test
   public void testCannotAddNullSolrInputDocument() {
+    exception.expect(NullPointerException.class);
+    exception.expectMessage("Cannot add a null SolrInputDocument");
+
     UpdateRequest req = new UpdateRequest();
     req.add((SolrInputDocument) null);
   }
 
   @Test
   public void testCannotAddNullDocumentWithOverwrite() {
+    exception.expect(NullPointerException.class);
+    exception.expectMessage("Cannot add a null SolrInputDocument");
+
     UpdateRequest req = new UpdateRequest();
     req.add(null, true);
   }
 
   @Test
   public void testCannotAddNullDocumentWithCommitWithin() {
+    exception.expect(NullPointerException.class);
+    exception.expectMessage("Cannot add a null SolrInputDocument");
+
     UpdateRequest req = new UpdateRequest();
     req.add(null, 1);
   }
 
   @Test
   public void testCannotAddNullDocumentWithParameters() {
+    exception.expect(NullPointerException.class);
+    exception.expectMessage("Cannot add a null SolrInputDocument");
+
     UpdateRequest req = new UpdateRequest();
     req.add(null, 1, true);
   }
 
   @Test
   public void testCannotAddNullDocumentAsPartOfList() {
+    exception.expect(NullPointerException.class);
+    exception.expectMessage("Cannot add a null SolrInputDocument");
+
     UpdateRequest req = new UpdateRequest();
     req.add(Arrays.asList(new SolrInputDocument(), new SolrInputDocument(), null));
   }
 
+  @Test
+  public void testEqualsMethod() {
+    final SolrInputDocument doc1 = new SolrInputDocument("id", "1", "value_s", "foo");
+    final SolrInputDocument doc2 = new SolrInputDocument("id", "2", "value_s", "bar");
+    final SolrInputDocument doc3 = new SolrInputDocument("id", "3", "value_s", "baz");
+/*
```

Comment:
left over from other testing?

Reply:
Yes.  My bad - the tests are a bit WIP/experimental at this point.  I ran out of weekend-time partially into starting them.  I made a comment about this in the jira, but should've mentioned it here where it'd come more easily to the attention of reviewers.